src/zope/proxy/__init__.py

```diff
@@ -251,16 +251,15 @@ def __iter__(self):
         # time.
         return iter(self._wrapped)
 
-    def next(self):  # pragma: no cover PY2
+    def next(self):
         # Called when we wrap an iterator itself.
-        return self._wrapped.next()
+        return next(self._wrapped)
 
     def __next__(self):
         return self._wrapped.__next__()
 
-    # Python 2.7 won't let the C wrapper support __reversed__ :(
-    # def __reversed__(self):
-    #    return reversed(self._wrapped)
+    def __reversed__(self):
+        return reversed(self._wrapped)
```

Comment:
Note to self: check that the C wrapper supports it, and that this is mentioned in the changelog.

Reply:
I commented the code again and added a note. This research should be done separately, right now I am just doing the mechanical job of cleaning out what I can.